### PR TITLE
[csharp] Address obsolete build warnings in ApiClient

### PR DESCRIPTION
**Pull request by alec-petersen**
See issue #19525 

- Non-async overload of `Deserialize` is obsolete, RestSharp suggests using async overload of `Deserialize` instead.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming `7.x.0` minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
appveyor tests passed via https://ci.appveyor.com/project/WilliamCheng/openapi-generator/builds/50782894

File: modules/openapi-generator/src/main/resources/csharp/ApiClient.mustache [hunk 1]
```diff
@@ -455,7 +455,7 @@ namespace {{packageName}}.Client
             var clientOptions = new RestClientOptions(baseUrl)
             {
                 ClientCertificates = configuration.ClientCertificates,
-                MaxTimeout = configuration.Timeout,
+                Timeout = TimeSpan.FromMilliseconds(configuration.Timeout),
```
[the remark this answers is not in the document]
Comment (wing328) [reply modules/openapi-generator/src/main/resources/csharp/ApiClient.mustache:458]: thanks for the PR

looks like a breaking change

can we keep the current behaviour for the time being?

**Comment by wing328** [comment]
cc 
@mandrean (2017/08) @shibayan (2020/02) @Blackclaws (2021/03) @lucamazzanti (2021/05) @iBicha (2023/07)

File: modules/openapi-generator/src/main/resources/csharp/ApiClient.mustache (hunk 1 above)
[the remark this answers is not in the document]
Comment (alec-petersen) [reply modules/openapi-generator/src/main/resources/csharp/ApiClient.mustache:458]: @wing328 That's fine with me, are you suggesting undoing the change for this line and opening a new PR for 8.0.0?

**Comment by alec-petersen** [comment]
@wing328 Undid the timeout change, I'll open a PR to 8.0.0 for that

**Pull request merged**